libsolidity/codegen/Compiler.cpp

```diff
@@ -37,16 +37,6 @@ void Compiler::compileContract(
 	bytes const& _metadata
 )
 {
-	auto static notTransient = [](VariableDeclaration const* _varDeclaration) {
-		solAssert(_varDeclaration);
-		return _varDeclaration->referenceLocation() != VariableDeclaration::Location::Transient;
-	};
-
-	solUnimplementedAssert(
-		ranges::all_of(_contract.stateVariables(), notTransient),
-		"Transient storage variables are not supported."
-	);
-
```

Comment:
Maybe keep asserting that we don't have non-value-type transient storage variables.

Reply:
Done.